sncast_std/src/lib.cairo

```diff
@@ -5,14 +5,97 @@ use core::fmt::{Debug, Display, Error, Formatter};
 
 #[derive(Drop, PartialEq, Serde, Debug)]
 pub struct ErrorData {
-    msg: ByteArray
+    revert_error: ContractExecutionError
 }
 
-#[derive(Drop, PartialEq, Serde, Debug)]
+#[derive(Drop, PartialEq, Debug)]
 pub struct TransactionExecutionErrorData {
     pub transaction_index: felt252,
-    // TODO(#3120): Implement `ContractExecutionError` and update below field type
-    pub execution_error: ByteArray,
+    pub execution_error: ContractExecutionError,
+}
+
+impl TransactionExecutionErrorDataSerde of Serde<TransactionExecutionErrorData> {
+    fn serialize(self: @TransactionExecutionErrorData, ref output: Array<felt252>) {
+        output.append(*self.transaction_index);
+        self.execution_error.serialize(ref output);
+    }
+    fn deserialize(ref serialized: Span<felt252>) -> Option<TransactionExecutionErrorData> {
+        let transaction_index = (*serialized.pop_front()?);
+        let execution_error = Serde::<ContractExecutionError>::deserialize(ref serialized).unwrap();
+        Option::Some(TransactionExecutionErrorData { transaction_index, execution_error })
+    }
+}
+
+
+// TODO(#3120): We should implement displaying message variant as deserialized byte array.
+// ATM, it's dispplayed as a serialized one.
+#[derive(Drop, PartialEq, Copy, Debug)]
+pub enum ContractExecutionError {
+    Nested: Box<ContractExecutionErrorInner>,
+    Message: Span<felt252>,
```

Comment:
note: We can't use `ByteArray` for `Message`, because `ContractExecutionError` must derive `Copy`.